.github/workflows/ci.yml

```
@@ -73,3 +80,85 @@
+          cache: "pnpm"
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: 🔍 Build Validation
+        run: |
+          echo "🔍 Running comprehensive build validation..."
+          pnpm run validate
+
+      - name: 📝 Commit Message Validation
+        run: |
```

actionlint (1.7.7)

"github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions for more details

(expression)